pkg/solana/chainwriter/chain_writer.go

```diff
@@ -0,0 +1,324 @@
+package chainwriter
```

Comment:
Are we planning to rename it later to [ContractWriter](https://github.com/smartcontractkit/chainlink-common/blob/db7919d60550c76b37a0d7cc5e694a272cc54bdd/pkg/types/contract_writer.go#L14)? Maybe it's easier to do it now once and for all for Solana? 

I know it's a very annoying nit 🙏

Reply:
To be honest, I don't really understand the distinction between ContractWriter and ChainWriter, maybe @ilija42 can fill in here. My understanding was that the chain-specific version would still be called ChainWriter and the generalized interface is the ContractWriter, but I'm not sure if that's correct nor do I have strong opinions on the semantics either way.

Reply:
Maybe [this](https://github.com/smartcontractkit/chainlink-solana/pull/935#pullrequestreview-2520580600) answers our question about what to do 😅

Reply:
> My understanding was that the chain-specific version would still be called ChainWriter and the generalized interface is the ContractWriter, but I'm not sure if that's correct nor do I have strong opinions on the semantics either way.

The Relayer interface should have ChainWriter and ChainReader, which consist of ContractWriter and ContractReader. Contract Reader/Writer are components that need to be configured and initialised, while the Chain Reader/Writer interfaces have functionalities that don't. Some examples are `LatestHead`, `GetChainStatus` for Reading and `Transact` for Writing. It would be nice to be consistent with naming across all Relayer implementations.